app/models/dish.rb

```diff
@@ -1,4 +1,10 @@
 class Dish < ApplicationRecord
   validates_presence_of :name, :description
   belongs_to :chef
+  has_many :dish_ingredients
+  has_many :ingredients, through: :dish_ingredients
+
+  def calories
+  self.ingredients.pluck(:calories).sum
+  end
```

Comment:
The `#sum` method that you are using at the end, although not a bad use of it is actually the Array `sum` method and not AR. In this scenario, it's not bad, but I just wanted to point that out for future reference. The AR `sum` method takes a mandatory argument of the column that needs to be summed.

```ruby
def calories
  self.ingredients.sum(:calories)
end
```